atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/atomix/AtomixTimestampServiceIntegrationTest.java

```diff
@@ -60,44 +61,57 @@ public static void stopAtomix() {
     @Before
     public void setupTimestampService() {
         DistributedLong distributedLong = DistributedValues.getTimestampForClient(ATOMIX_REPLICA, CLIENT_KEY);
-        timestampService = new AtomixTimestampService(distributedLong);
+        atomixTimestampService = new AtomixTimestampService(distributedLong);
     }
 
     @Test
-    public void timestampsShouldBeIncreasing() {
-        long ts1 = timestampService.getFreshTimestamp();
-        long ts2 = timestampService.getFreshTimestamp();
-
-        assertThat(ts1).isLessThan(ts2);
+    public void timestampsAreReturnedInOrder() {
+        TimestampServiceTests.timestampsAreReturnedInOrder(atomixTimestampService);
     }
 
     @Test
     public void canRequestTimestampRange() {
-        int expectedNumTimestamps = 5;
-        TimestampRange range = timestampService.getFreshTimestamps(expectedNumTimestamps);
-
-        long actualNumTimestamps = range.getUpperBound() - range.getLowerBound() + 1;
-        assertThat(actualNumTimestamps)
-                .withFailMessage("Expected %d timestamps, got %d timestamps. (The returned range was: %d-%d)",
-                        expectedNumTimestamps, actualNumTimestamps, range.getLowerBound(), range.getUpperBound())
-                .isEqualTo(expectedNumTimestamps);
+        TimestampServiceTests.canRequestTimestampRangeWithGetFreshTimestamps(atomixTimestampService);
+    }
+
+    @Test
+    public void timestampRangesAreReturnedInNonOverlappingOrder() {
+        TimestampServiceTests.timestampRangesAreReturnedInNonOverlappingOrder(atomixTimestampService);
+    }
+
+    @Test
+    public void willNotHandOutTimestampsEarlierThanAFastForward() {
+        TimestampServiceTests.willNotHandOutTimestampsEarlierThanAFastForward(
```

Comment:
Discussed offline, leaving comment here for others' SA:
the problem with using an abstract class for `TimestampServiceTest(s)` was the desire to keep the `TimestampService` and `TimestampMigrationService` interfaces separate, but here we need `atomixTimestampService` to be an instance of both.

This can be done without loss of generality by making a new interface that extends both interfaces, and having `AtomixTimestampService` implement that.